src/colab/server-picker.ts

```diff
@@ -126,23 +163,56 @@ export class ServerPicker {
     if (!isAcceleratorDefined(state)) {
       return;
     }
+    const shapePicks = getShapePicksFromAccelerator(
+      shapesByAccelerators,
+      state.accelerator,
+    );
+    if (shapePicks.length <= 1) {
+      state.shape =
+        shapePicks.length === 1 ? shapePicks[0].value : Shape.STANDARD;
+      return (input: MultiStepInput) =>
+        this.promptForAlias(input, state, /** totalSteps= */ 3);
+    }
+    return (input: MultiStepInput) =>
+      this.promptForMachineShape(input, state, shapePicks);
+  }
+
+  private async promptForMachineShape(
+    input: MultiStepInput,
+    state: PartialServerWith<'variant'>,
+    items: ShapePick[],
+  ) {
+    const step = state.accelerator && state.accelerator !== 'NONE' ? 3 : 2;
+    const pick = await input.showQuickPick({
+      title: 'Select a machine shape',
+      step,
+      totalSteps: step + 1,
```

Comment:
Nit: Consider defining `const totalSteps = step + 1` below the definition of `step` to be used here and when calling `promptForAlias` (then we don't need the inline param comment).